### PR TITLE
add context to SubscriberOptions

### PR DESCRIPTION
**Pull request by vtolstov**
Signed-off-by: Vasiliy Tolstov <v.tolstov@unistack.org>

### CODE REVIEW

File: broker/options.go [hunk 1]
```diff
@@ -111,3 +111,10 @@ func TLSConfig(t *tls.Config) Option {
 		o.TLSConfig = t
 	}
 }
+
+// Context set context
+func Context(ctx context.Context) SubscribeOption {
```
Comment (asim) [inline broker/options.go]: Considering this is for Subscribe it should be SubscribeContext

File: server/handler.go [hunk 2]
```diff
@@ -1,13 +1,30 @@
...
+import "context"
+
 type HandlerOptions struct {
 	Internal bool
 	Metadata map[string]map[string]string
 }
 
+type SubscriberOption func(*SubscriberOptions)
+
+func NewSubscriberOptions(opts ...SubscriberOption) SubscriberOptions {
```
Comment (asim) [inline server/handler.go]: Move to bottom of file

**Comment by vtolstov** [comment]
i'm fix issues

File: server/handler.go [hunk 3]
```diff
@@ -35,9 +34,36 @@ func InternalSubscriber(b bool) SubscriberOption {
...
+	}
+
+	for _, o := range opts {
+		o(&opt)
+	}
+
+	return opt
+}
+
+type SubscriberOptions struct {
```
Comment (asim) [inline server/handler.go]: Types should remain at the top. Only move NewSubscriberOptions.

File: server/rpc_server.go [hunk 4]
```diff
@@ -357,6 +357,9 @@ func (s *rpcServer) Register() error {
 		if queue := sb.Options().Queue; len(queue) > 0 {
 			opts = append(opts, broker.Queue(queue))
 		}
+		if cx := sb.Options().Context; cx != nil {
+		opts = append(opts, broker.Context(cx))
```
Comment (asim) [inline server/rpc_server.go]: Please `go fmt`. Also broker.Context is no longer valid.

File: server/handler.go (hunk 3 above)
Comment (vtolstov) [reply server/handler.go]: ok

**Pull request merged**